pep-0458-tuf-online-keys.rst

```diff
@@ -499,16 +701,143 @@ SHOULD also be made available to clients.
+verify cryptographic signatures). Verification of signatures MUST be done in
+Python, and verifying RSA [20]_ signatures in pure-Python may be impractical
+due to speed. Therefore, PyPI SHOULD use the Ed25519 signature scheme.
+
+
+Cryptographic Key Files
+-----------------------
+
+Cryptographic keys MAY be stored in password-protected, encrypted key files.
+Administrators MAY use the repository tool to encrypt key files with
+AES-256-CTR-Mode and strengthen passwords with PBKDF2-HMAC-SHA256 (100K
+iterations by default, but this may be overridden by the developer). The
+current Python implementation of TUF can use any cryptographic library
+(PyCrypto [24]_ is currently used to encrypt the TUF key files, but support for
+PyCA Cryptography can be added in the future), may override the default number
+of PBKDF2 iterations, and the KDF tweaked to preference.  However, the exact
+cryptographic constructions can be adjusted to include future primitives added
+by the cryptographic libraries supported.
+
+
+Keys objects stored in encrypted key files and in metadata have the format:
+
+{ "keytype" : KEYTYPE,
+  "keyval" : KEYVAL
+}
+
+All keys have the format:
+
+{ "keytype" : KEYTYPE,
+"keyval" : KEYVAL }
+
+where KEYTYPE is a string describing the type of the key ("ed25519") and how
+it's used to sign documents.  The type determines the interpretation of KEYVAL.
+
+The 'ed25519' key format is:
+
+{ "keytype" : "ed25519",
+  "keyval" :
+    { "public" : PUBLIC,
+      "private" : PRIVATE
+    }
+}
+
+where PUBLIC and PRIVATE are both 32-byte strings.
+
+Metadata does not include the private portion of the key object:
+
+{ "keytype" : "ed25519",
+  "keyval" :
+    { "public" : PUBLIC}
+}
+
+The KEYID of a key is the hexdigest of the SHA-256 hash of the canonical JSON
+form of the key, where the "private" object key is excluded.
+
+
+Generating Cryptographic Keys and Signing Metadata
+--------------------------------------------------
+
+The repository management tool may be used to generate the cryptographic keys
+and sign the PyPI metadata downloaded by end users.  The following Python code
+demonstrates how to generate and import cryptographic keys with the repository
+management tool:
+
+.. code-block::
+
+  >>> from tuf.repository_tool import *
+
+  # Generate and write an ed25519 key pair.  The private key is saved encrypted.
+  # A 'password' argument may be supplied, otherwise a prompt is presented.
+  >>> generate_and_write_ed25519_keypair('/path/to/ed25519_key')
+  Enter a password for the ED25519 key: 
+  Confirm:
+
+  # Import the ed25519 public key just created . . .
+  >>> public_ed25519_key = import_ed25519_publickey_from_file('/path/to/ed25519_key.pub')
+
+  # and its corresponding private key.
+  >>> private_ed25519_key = import_ed25519_privatekey_from_file('/path/to/ed25519_key')
+  Enter a password for the encrypted ED25519 key: 
+
+
+The repository tool can use the cryptographic keys that are imported to sign
```

Comment:
800: do you mean to say "use the cryptographic keys that are imported to sign" or, could you instead say (better): "use the imported cryptographic keys to sign"